### ISSUE TITLE
com.ichi2.anki.exception.UnknownHttpResponseException: Internal Server Error

### ISSUE THREAD

**Issue by colonelsmoothie** [dsnopek/anki-sync-server#54]
Hello,

I've been experiencing some odd behavior regarding the sync on AnkiDroid. Sometimes when I sync, "Writing changes into database..." will hang at 1459 kB for about 3 minutes, after which I get the error message:

![screenshot_20170108-211540](https://cloud.githubusercontent.com/assets/3887707/21756940/428cf3be-d5ed-11e6-85e7-c912c17db69a.png)

![screenshot_20170108-132456](https://cloud.githubusercontent.com/assets/3887707/21756941/45dc33ea-d5ed-11e6-8d68-673d771a0ffd.png)

The cards seem to update, but the media doesn't. I tried syncing against an empty folder, but only about 1/3rd of the media downloaded, then I got the error message above. This issue is only happening on my phone. Desktop-to-desktop syncs work fine.

I did a hard reboot of my phone and I was able to redownload all the media files. I then added some media to my desktop collection to see if I could sync it to my Android phone. The syncing screen took a minute, but I was able to successfully sync the media. I'll see if I can reproduce the issue.

**Comment by ninja33** [comment]
It's not anki server's problem, it's ankidroid's bug.
please check #50, and it should be fixed by ankidroid next release 2.8

**Comment by colonelsmoothie** [comment]
I see. In the meantime, do you think there's anything I can do to make life easier? Has it been fixed in AnkiDroid 2.8 beta or should I just wait until it gets released?

**Comment by ninja33** [comment]
You can go to ankidroid github download 2.8 beta3 
https://github.com/ankidroid/Anki-Android/releases/tag/v2.8beta3

**Comment by colonelsmoothie** [comment]
I read the bug report you showed me, and I was able to reproduce the issue by emptying some unused media, so that's consistent with the report.

I installed the beta. Syncing works, I tried doing various things like adding media, LaTeX, and also deleting media, all of which worked. My only issue is that syncing is slow whenever there are media changes (3-5 min). But at least it syncs correctly. Thanks.
